### ISSUE TITLE
Add support of syllable lyrics

### ISSUE THREAD

**Issue by dm-vodopyanov** [tranxuanthang/lrcget#51]
Syllable lyrics is an extension of enhanced LRC format, for more see here: https://en.wikipedia.org/wiki/LRC_(file_format)#A2_extension:_word_time_tag

Example:

```
[00:00.00] <00:00.04> When <00:00.16> the <00:00.82> truth <00:01.29> is <00:01.63> found <00:03.09> to <00:03.37> be <00:05.92> lies 
[00:06.47] <00:07.67> And <00:07.94> all <00:08.36> the <00:08.63> joy <00:10.28> within <00:10.53> you <00:13.09> dies 
[00:13.34] <00:14.32> Don't <00:14.73> you <00:15.14> want <00:15.57> somebody <00:16.09> to <00:16.46> love
```

The request is add an ability to differentiate syllable lyrics from usual "Synced" LRC files in LRCGET and correctly display them in LRCGET.

**Comment by Tox1cPhantom** [comment]
@tranxuanthang Is there a way to get Syllable lyrics?

**Comment by tranxuanthang** [comment]
@Tox1cPhantom There isn't one for now. I'm still not so sure if it is worth it, because word-by-word sync will be much more time-consuming, so the contribution rate to this format might be extremely low.

But who knows, if there is high demand for this, then it will be added to LRCLIB and LRCGET someday.

**Comment by dm-vodopyanov** [comment]
@tranxuanthang there is no need to create syllable lyrics manually, one can download syllable lyrics from Apple Music in TTML format and convert them into syllable LRC lyrics.

**Comment by ukrbugreporter** [comment]
> @Tox1cPhantom There isn't one for now.

[OneTagger](https://github.com/Marekkon5/onetagger) can fetch syllable lyrics in .LRC format from Musixmatch. The dev [mentioned](https://github.com/Marekkon5/onetagger/issues/211#issuecomment-1500886920) using some "hidden APIs" for this, but it actually works.

**Comment by mihawk90** [comment]
In the same vein, if this is ever implemented, LRCGET should recognise `.elrc` files for this purpose.

Should also be noted LRCGET will let you publish lyrics in that format:

<img width="428" height="244" alt="Image" src="https://github.com/user-attachments/assets/ac9b24f8-5072-4b11-9ecf-544670cf9a09" />

Just not search for them.

**Comment by mixtapejaxson** [comment]
> [@Tox1cPhantom](https://github.com/Tox1cPhantom) There isn't one for now. I'm still not so sure if it is worth it, because word-by-word sync will be much more time-consuming, so the contribution rate to this format might be extremely low.
> 
> But who knows, if there is high demand for this, then it will be added to LRCLIB and LRCGET someday.

@Tox1cPhantom I have an open source app on flathub that I download which makes it painlessly easy. I'd love to see this get more adopted and I'd gladly contribute (I've already converted most of my music)

**Comment by mihawk90** [comment]
> But who knows, if there is high demand for this, then it will be added to LRCLIB and LRCGET someday.

This also sounds like a chicken and egg situation. Adoption in lrclib won't happen when there is no way to search for them, which means there's no real way to measure demand. GitHub issues are a poor way to track demand as a lot of users will not be using GitHub in the first place. Most users likely only use lrcget or some other automated downloading tool.

Should also note there is an equivalent request on the lrcljb repo:
* tranxuanthang/lrclib#8
* tranxuanthang/lrclib#78

So demand is definitely there.

> I have an open source app on flathub that I download which makes it painlessly easy.

I assume you're talking about ~~Phonograph~~ Chronograph? It's a neat tool, but from what I could tell it has the same problem lrcget itself has: the seeking and "just time the click right" is too imprecise, especially for word-by-word timing.
I might open a feature request eventually but what I'd like to see is a spectrum analyzer akin to what Aegisub provides to be able to do precise timing. I've been using Aegisub for this every now and then, with a script to export (e)LRC files.

**Comment by mixtapejaxson** [comment]
> > I have an open source app on flathub that I download which makes it painlessly easy.
> 
> I assume you're talking about Phonograph? It's a neat tool, but from what I could tell it has the same problem lrcget itself has: the seeking and "just time the click right" is too imprecise, especially for word-by-word timing. I might open a feature request eventually but what I'd like to see is a spectrum analyzer akin to what Aegisub provides to be able to do precise timing. I've been using Aegisub for this every now and then, with a script to export (e)LRC files.

@mihawk90 
Yes. The one I have is [Chronograph](https://flathub.org/en/apps/io.github.dzheremi2.lrcmake-gtk). You are right it is a bit imprecise. After using it for about a month I can say it's not the greatest. I have plans to create a truely FOSS lyric sync creator. However that's a lot of work but hopefully it will lead to *(one day)* eventually having word-by-word syncing for LRCLIB & LRCGet.

**Comment by mihawk90** [comment]
> I have plans to create a truely FOSS lyric sync creator.

<img width="1000" height="567" alt="Image" src="https://github.com/user-attachments/assets/4a0c62b5-4199-4550-900e-83abc3ba1df2" />

**Comment by ArjixWasTaken** [comment]
So syllable/word sync is already supported or is it an edge case to encounter such?

https://github.com/pear-devs/pear-desktop/issues/4265#issuecomment-3788782098

**Comment by mihawk90** [comment]
They can be submitted because they pass the validation RegEx, but they can't be searched for, which is the point of this request.

**Comment by ArjixWasTaken** [comment]
> They can be submitted because they pass the validation RegEx

Is it intentional that they pass the validation regex?

**Comment by mihawk90** [comment]
Don't know but the RegEx is very simple as it basically just checks that the timestamp format is the same for every line:
https://github.com/tranxuanthang/lrcget/blob/3084a7f4f9c820a7d452ef476b5232bba50f350e/src/utils/lyrics.js#L6-L33

Should be noted that this is *client*-side and I don't know whether an equivalent validation exists on the server side. Other applications may or may not choose to enforce the same or different formats.

It would also be questionably useful to filter out anyway. The additional timestamps are either used by applications or sanitised out, and if not then users can just choose an alternative from lrclib given how many duplicates any given track has.

I think the only way forward *is* making them discoverable because a) they already exist and b) as noted above there isn't really any other way to gauge demand when there's no way to measure said demand because they aren't discoverable.

**Comment by AARP41298** [comment]
i think https://github.com/tranxuanthang/lrclib/issues/39 is related

**Comment by AARP41298** [comment]
s. idea: emojis instead of walaoke

<img width="349" height="290" alt="Image" src="https://github.com/user-attachments/assets/cee9a4db-3a4c-4802-b1f4-696a3bda61f3" />

**Comment by tranxuanthang** [comment]
LRCGET and LRCLIB will adopt new Lyricsfile format with support for word-by-word sync!

Check out v2.0.0 now: https://github.com/tranxuanthang/lrcget/releases/tag/2.0.0

**Comment by ArjixWasTaken** [comment]
Amazing, one format to rule them all! (I hope)

If you want to see massive adoption, I suggest you publish an npm package with typescript types for it, and maybe a validation layer using zod?

It's not a hard format to parse, ofc, but having an official "sdk" (not really an sdk but you get the point) will make it easier to recommend to others, spread visibility, etc

And since you plan to have breaking changes, you could maybe version the schema, and provide automatic conversion to the latest version.

(if you are already working on smth like this, then excuse me)

**Comment by tranxuanthang** [comment]
Close this since this issue has been resolved!
